src/plugins/discover/public/plugin.tsx

```diff
@@ -302,27 +302,41 @@ export class DiscoverPlugin
     }
   }
 
-  private createProfileServices = once(async ({ plugins }: { plugins: DiscoverStartPlugins }) => {
-    const { registerProfileProviders } = await import('./context_awareness/profile_providers');
+  private createProfileServices() {
     const rootProfileService = new RootProfileService();
     const dataSourceProfileService = new DataSourceProfileService();
     const documentProfileService = new DocumentProfileService();
+
+    return { rootProfileService, dataSourceProfileService, documentProfileService };
+  }
+
+  private createProfilesManager = once(async (core: CoreStart, plugins: DiscoverStartPlugins) => {
```

Comment:
These changes break context awareness in dashboards. Moving `once` to `createProfilesManager` makes the profiles manager a singleton, but each instance of Discover or the saved search embeddable should get its own profiles manager instance since it's tied to the application state. Otherwise the context awareness state of one panel gets mangled when another panel is added, like this:

https://github.com/user-attachments/assets/a7eb13e9-d54e-46e5-835d-7ae31f3217b3

The profile services were safe to be singletons since we only needed to register providers once on initialization and could share them between profiles manager instances, although I'm not sure if that's really the case now either since we're now passing dependencies to providers.

Reply:
Good catch! I fixed it removing the memoization so that a new Profiles manager is created for each instance of the embeddable or usage in Discover.
I also agree that there is not such a performance impact in creating new profile services when required vs memorizing them, compared to the risk of undesired side effects as happened here.
I would go for not having this memo in place atm.

Here is the embeddable working as expected:

https://github.com/user-attachments/assets/9b4fd1dc-ffba-44de-9d6a-758f3f9aab7d